conans/test/functional/toolchain/test_msbuild.py

```diff
@@ -369,11 +361,12 @@ def test_toolchain_win_debug(self):
 
         # Run the configure corresponding to this test case
         client.run("install . %s -if=conan" % (settings, ))
-        self.assertIn("conanfile.py: MSBuildToolchain created conan_toolchain_debug_x64_v140.props",
+        self.assertIn("conanfile.py: MSBuildToolchain created conan_toolchain_debug_x64.props",
                       client.out)
         vs_path = vs_installation_path("15")
         vcvars_path = os.path.join(vs_path, "VC/Auxiliary/Build/vcvarsall.bat")
 
+        # FIXME: This is cheating, pass the toolset on the command line, nothing that devs would do
```

Comment:
I am not sure I understand what is cheating here? I guess it's legal to pass toolset on command line, and developers can do that, correct?

Reply:
99% of developers will double click on the solution to open it, not build it in the command line with ``msbuild`` command, don't you think?

Reply:
@solvingj WDYT?
I understand, it might not be common for majority of users, but still possible and legal.
still I believe most of CI scripts invoke msbuild or devenv, so it should be supported case IMO.

Reply:
It will be managed by the toolchain, so still very possible in CI. The only thing is that it will not be possible to build different toolset configurations in the same run, with the same build. You can do different ``conan install`` with different toolsets and still build with them, just not build sequentially for different toolsets.

What we are really trying to avoid as much as possible here is to have something that is done only in CI, but developers cannot easily replicate.  The problem is that if we add the toolset variability to the generated files, we need the build helper to pass the toolset definition in the command line. If the build helper passes that definition in the command line, then developers are no longer able to reproduce the build from their IDE, which is a major goal for this integration approach. We cannot have a build from the ``build()`` method that a developer cannot achieve as a consumer, with their native tools.

Reply:
Yes, 2.1: the flow is to add the properties file to the visual project. This can be done from the IDE or editing the .vcxproj. I think this flow has several advantages:

- It is completely explicit, no magic.
- The addition of the .props file needs to be done only once, as the .vcxproj is typically commited to source control
- It can be done without command line (besides Conan commands). Note that typically ``msbuild`` is NOT in the path, so developer needs to open a matching VS-prompt. More complicated than double click on the .sln

From our experience Microsoft and Visual Studio developers are extremely reluctant to use command line, and this is already complicated for Conan adoption. Companies sysdmins/devops/build-responsibles are going to large extends to avoid their developers to touch the terminal.

We should optimize for user stories that are as close to the current dev experience as possible. I see a common user story in Msft/VS devs is:
- Get the source code, with git clone or similar
- Double click the .sln to open the project

Here, we aim to only introduce an extra step, the ``conan install`` of dependencies. Some will use MSBuild tasks or plugins, others will add a .bat file in the root to double click (that contains the conan install).

In the cmake user story, it is more like:
- Get the source code, git clone
- cmake .. something something # to create the project
- OpenIDE, editor, build command line...

So the user is already doing the ``cmake`` command, it is very necessary, a majority of cmake users will call cmake themselves to generate the project. In this case, we should optimize so users can keep doing the ``cmake`` command as close as their current one as possible.

Reply:
okay, in that case, if I open solution and inject `conan.props` (via PropertyManager or manually), it will update `RuntimeLibrary` accordingly, right?
same way, I could edit property file via PropertyManager (or via text editor) and update the `RuntimeLibrary` to the value I need.
everything without command line, only via IDE.

Reply:
Yes, as the ``conan_toolchain.props`` is injected, doing a ``conan install ... -s compiler.runtime=MT`` or similar will update the toolchain and it will be taken into account.

> same way, I could edit property file via PropertyManager (or via text editor) and update the RuntimeLibrary to the value I need.

It depends on the evaluation order. It seems that including the toolchain in the right position, it will have priority and define the final value of the toolset.